src/pages/Order/index.tsx

```diff
@@ -0,0 +1,185 @@
+import {
+  Box,
+  Button,
+  ChakraProvider,
+  Checkbox,
+  Flex,
+  HStack,
+  Image,
+  Input,
+  Select,
+  Text,
+  Textarea,
+  VStack,
+} from '@chakra-ui/react';
+import { useRef, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+
+export const Order = () => {
+  const location = useLocation();
+  const { productDetail, quantity, price } = location.state || {};
+
```

Comment:
```
export interface OrderLocationState {
  productDetail: ProductDetail;
  quantity: number;
  price: number;
}
```

location.state는 기본적으로 타입이 any라서 타입 안전하게 소비하기 위해 26번 라인 위에 location state 타입을 선언해두고 타입을 명시해주는게 좋겠습니다.

```suggestion
  const { productDetail, quantity, price } = (location.state || {}) as OrderLocationState;
```